### PR TITLE
homework2.1

### PR DESCRIPTION
**Pull request by jfdsmit**

### CODE REVIEW

File: course5/homework2.1/carmack.cc [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+#include <iostream>
+#include <cstdlib>
+
+const float ONE_HALF = 0.5;
+const float THREE_HALFS = 1.5;
+const int FAST_INVERSE_SQRT_MAGIC = 0x5f3759df;
+
+float fastInvSqrt(float input)
+{
+        unsigned long temporaryIntegerForm;
```
Comment (maciekgajewski) [inline course5/homework2.1/carmack.cc]: It is essential that the integer is 32-bits. To achieve portable result, use ```std::uint32_t```

File: course5/homework2.1/carmack.cc [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+float fastSqrt(float input)
+{
+        return 1 / fastInvSqrt(input);
+}
+
+int main(int argc, char** argv)
+{
+        for (auto i = 1; i < argc; i++)
+        {
+                auto input = std::atof(argv[i]);
```
Comment (maciekgajewski) [inline course5/homework2.1/carmack.cc]: Use ```std::stof``` instead. ```atof``` is na old, C function, and it reports error by return 0.0. stof throws exception, and hence is more explicit and safer.

**Review by maciekgajewski — COMMENTED** [review]
Very good!

File: course5/homework2.1/carmack.cc (hunk 2 above)
Comment (jfdsmit) [reply course5/homework2.1/carmack.cc]: Ah, that's the difference, thanks :)

File: course5/homework2.1/carmack.cc (hunk 1 above)
Comment (jfdsmit) [reply course5/homework2.1/carmack.cc]: Very right, good point

**Pull request closed without merging**